src/views/Home/utils.ts

```diff
@@ -0,0 +1,148 @@
+import { isDefined, getDuplicates, isNotDefined, listToMap } from '@togglecorp/fujs';
+
+import {
+    Settings,
+    Code,
+    AdminLevel,
+} from './typings';
+
+interface Property {
+    index: number;
+    code?: Code;
+    name?: string;
+    parentCode?: Code;
+    // parentName?: string;
+}
+interface Error {
+    title: string;
+    severity: 'error' | 'warning';
+    description?: string;
+}
+
+function getProperties(settings: Settings) {
+    const {
+        geoJson,
+        pointer,
+    } = settings;
+
+    return geoJson.features.map((feature, index) => ({
+        index,
+        code: isDefined(pointer.code)
+            ? (feature.properties[pointer.code]) as Code | undefined
+            : undefined,
+        name: isDefined(pointer.name)
+            ? (feature.properties[pointer.name]) as string | undefined
+            : undefined,
+        parentCode: isDefined(pointer.parentCode)
+            ? (feature.properties[pointer.parentCode]) as Code | undefined
+            : undefined,
+        /*
+        parentName: isDefined(pointer.parentName)
+            ? (feature.properties[pointer.parentName]) as string | undefined
+            : undefined,
+        */
+    }));
+}
+
+function validateSettings(settings: Settings, root = false): Error[] {
+    function getErrors(
```

Comment:
why is this inside?

Reply:
Because this is not used anywhere else. Private function!

Reply:
humm, not a big fan of private functions.
We should share.